app/assets/src/components/common/MetadataCSVLocationsMenu.jsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { uniq, find } from "lodash/fp";
+
+import { withAnalytics, logAnalyticsEvent } from "~/api/analytics";
+import { getGeoSearchSuggestions } from "~/api/locations";
+import MetadataInput from "~/components/common/MetadataInput";
+import { processLocationSelection } from "~/components/ui/controls/GeoSearchInputBox";
+import PropTypes from "~/components/utils/propTypes";
+import DataTable from "~/components/visualizations/table/DataTable";
+
+import cs from "./metadata_csv_locations_menu.scss";
+
+// Batch geosearch CSV locations for matches
+export const geosearchCSVlocations = async (metadata, metadataType) => {
+  if (!(metadata && metadata.rows)) return;
+
+  // For each unique plain text value, get the #1 search result, if any.
+  const rawNames = uniq(metadata.rows.map(r => r[metadataType.name]));
+  const matchedLocations = {};
+  const requests = rawNames.map(async query => {
+    const res = await getGeoSearchSuggestions(query, 1);
+    if (res.length > 0) matchedLocations[query] = res[0];
+  });
+  await Promise.all(requests);
```

Comment:
Can you have request limit issues?

Reply:
it's possible yeah.. is there a way I should catch errors while using Promise.all?

Reply:
I have not used it... but I believe the promise will have `then` functions that will return an array of results (success) and a `catch` that will be called for each(or first?) error. This pattern on the web seems interesting: 
```
Promise.all([
  p1.catch(error => { return error }),
  p2.catch(error => { return error }),
]).then(values => { 
  console.log(values[0]) // "p1_delayed_resolution"
  console.log(values[1]) // "Error: p2_immediate_rejection"
})
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

I guess you might be able to guess when you are going to have the issue depending on how many promises you have and introduce some random delays to desync the requests?

Reply:
👍 added batching 20 requests at a time as discussed offline